pkg/controllermanager/controller/shoot/shoot_control_reconcile.go

```diff
@@ -370,6 +371,7 @@ func (c *defaultControl) updateShootStatusReconcileSuccess(o *operation.Operatio
 
 	newShoot, err = kutil.TryUpdateShootStatus(c.k8sGardenClient.Garden(), retry.DefaultRetry, newShoot.ObjectMeta,
 		func(shoot *gardenv1beta1.Shoot) (*gardenv1beta1.Shoot, error) {
+			helper.SetShootStatusKubeProxyMode(shoot)
```

Comment:
Why do we need this code? Let's remove it and include it with the PR that incorporates the cleanup. In this PR we should only prevent switching mode for k8s >= 1.14

Reply:
Would be also great to see the actual used kube-proxy mode in the shoot status. E.g. for older cluster with kubernetes version ```< 1.14.1``` but it can be also removed if we don't want to show it in the status.

Reply:
What for? If we don't use it right now let's not add it. This is confusing.